src/api/db/migrate/20180405074538_change_default_charset_and_mysql57_defaults.rb

```diff
@@ -0,0 +1,45 @@
+class ChangeDefaultCharsetAndMysql57Defaults < ActiveRecord::Migration[5.1]
+  # mysql/mariadb has changed the default to utf8mb4, we need to update old instances
+  # to avoid to have specific sql code for difference mysql versions
+  def up
+    # these may be utf8 or utf8mb4 depending on the use mysql instance on creation time
+    [ 'ar_internal_metadata', 'binary_releases', 'bs_request_counter',
+      'cloud_azure_configurations', 'cloud_ec2_configurations', 
+      'cloud_user_upload_jobs', 'data_migrations', 'download_repositories',
+      'group_maintainers', 'incident_updateinfo_counter_values',
+      'kiwi_images', 'kiwi_package_groups', 'kiwi_packages',
+      'kiwi_preferences', 'kiwi_repositories', 'maintained_projects',
+      'product_media', 'product_update_repositories' ].each do |table|
+      execute("ALTER TABLE #{table} CONVERT TO CHARACTER SET utf8");
+    end
+    # contains free text, could have emojis
+    [ 'configurations', 'history_elements', 'kiwi_descriptions', 'notifications' ].each do |table|
+      execute("ALTER TABLE #{table} ROW_FORMAT=DYNAMIC"); # default in MySQL 5.7 & MariaDB 10.2
+      execute("ALTER TABLE #{table} CONVERT TO CHARACTER SET utf8mb4");
+    end
+    execute('alter table cloud_azure_configurations modify application_id mediumtext DEFAULT NULL;')
+    execute('alter table cloud_azure_configurations modify application_key mediumtext DEFAULT NULL;')
+    execute('alter table download_repositories modify pubkey mediumtext DEFAULT NULL;')
+    execute('alter table history_elements modify comment mediumtext DEFAULT NULL;')
+    execute('alter table notifications modify event_payload mediumtext NOT NULL;')
+  end
+
+  def down
+    # just normalize, it depends on which mysql instance the tables got created, so we can not
+    # really restore it
+    [ 'ar_internal_metadata', 'binary_releases', 'bs_request_counter',
+      'cloud_azure_configurations', 'cloud_ec2_configurations', 
+      'cloud_user_upload_jobs', 'configurations', 'data_migrations', 'download_repositories',
+      'group_maintainers', 'history_elements', 'incident_updateinfo_counter_values',
+      'kiwi_descriptions', 'kiwi_images', 'kiwi_package_groups', 'kiwi_packages',
+      'kiwi_preferences', 'kiwi_repositories', 'maintained_projects',
+      'notifications', 'product_media', 'product_update_repositories' ].each do |table|
+      execute("ALTER TABLE #{table} CONVERT TO CHARACTER SET utf8");
```

Comment:
Yes, but these tables should only contain strings were this is not possible (like project and package names). Also we did not support any product with the new databases, so all production installations should have utf8 here.

Reply:
tell if one of them has free text, we need to move them to the utf8mb then.

Reply:
> Yes, but these tables should only contain strings were this is not possible (like project and package names).  

See my emoji sample above. `HistoryElements` also include comments etc. etc. 

What's the problem with making this irreversible?

Reply:
history_elements is set to utf8mb4, so it should be fine?

you speak about rollback? well, it is hard to say what is causing more problems on old installations, using a data type of double size or cutting away some special chars in free text.

I picked utf8 because it was the former default, but no strong opinion here.

Reply:
ActiveRecord::IrreversibleMigration